tests/crud/test_hooks.py

```diff
@@ -0,0 +1,171 @@
+import json
+from unittest import TestCase
+
+from piccolo.columns import Integer, Varchar
+from piccolo.columns.readable import Readable
+from piccolo.table import Table
+from starlette.testclient import TestClient
+
+from piccolo_api.crud.endpoints import (
+    PiccoloCRUD,
+)
+from piccolo_api.crud.hooks import Hook, HookType
+
+
+class Movie(Table):
+    name = Varchar(length=100, required=True)
+    rating = Integer()
+
+    @classmethod
+    def get_readable(cls):
+        return Readable(template="%s", columns=[cls.name])
+
+
+async def set_movie_rating_10(row: Movie):
+    row.rating = 10
+    return row
+
+
+async def set_movie_rating_20(row: Movie):
+    row.rating = 20
+    return row
+
+
+async def remove_spaces(row_id: int, values: dict):
+    values["name"] = values["name"].replace(" ", "")
+    return values
+
+
+async def look_up_existing(row_id: int, values: dict):
+    row = await Movie.objects().get(Movie.id == row_id).run()
```

Comment:
Changing this line to `row = await Movie.objects().get(Movie._meta.primary_key == row_id).run()` should remove the type error.

Reply:
neat, thanks!